js/ui/windowManager.js

```diff
@@ -453,6 +453,140 @@ const AttachedZoomAnimation = new Lang.Class({
+                                                2.0,
+                                                1.0),
+        n_bounce: GObject.ParamSpec.uint('n-bounce',
+                                         'Number of bouces',
+                                         'Number of bounces',
+                                         GObject.ParamFlags.READWRITE |
+                                         GObject.ParamFlags.CONSTRUCT,
+                                         1,
+                                         10,
+                                         2),
+    },
+
+    vfunc_get_name: function() {
+        return 'bounce';
+    },
+
+    createActorPrivate: function(actor) {
+        let effect = new AttachedBounceAnimation({ bridge: this, enabled: false });
+        actor.add_effect_with_name('endless-animation-bounce', effect);
+        return effect;
+    }
+});
+
+// AttachedBounceAnimation
+//
+// A boune animation as attached to an actor.
+const AttachedBounceAnimation = new Lang.Class({
+    Name: 'AttachedBounceAnimation',
+    Extends: EndlessShellFX.Affine,
+    Implements: [ AnimationsDbus.ServerSurfaceAttachedEffect ],
+    Properties: {
+        'bridge': GObject.ParamSpec.object('bridge',
+                                           'Bounce Animation Bridge',
+                                           'The bridge to the settings',
+                                           GObject.ParamFlags.READWRITE |
+                                           GObject.ParamFlags.CONSTRUCT_ONLY,
+                                           ControllableBounceAnimation)
+    },
+
+    _init: function(params) {
+        this.parent(params);
+        this._enabledId = -1;
+    },
+
+    _startAnimation: function(animation, done) {
+        this.transform_animation = animation;
+        this.set_enabled(true);
+
+        this._enabledId = this.connect('notify::enabled', () => {
+            if (!this.enabled) {
+                done();
+                this.disconnect(this._enabledId);
+                this.enabledId = -1;
+            }
+        });
+    },
+
+    activate: function(event, detail) {
+        let [x, y] = this.actor.get_position();
+        let [width, height] = this.actor.get_size();
+
+        switch (event) {
+            case 'open':
+                this._startAnimation(new Animation.BounceAnimation({
+                    initial_scale: this.bridge.initial_scale,
+                    maximum_scale: this.bridge.maximum_scale,
+                    n_bounce: this.bridge.n_bounce,
+                    target: new Animation.Box({
+                        top_left: new Animation.Vector({ x: x, y: y }),
+                        bottom_right: new Animation.Vector({ x: x + width, y: y + height })
+                    }),
+                    stepper: new Animation.LinearStepper({ length: this.bridge.length })
+                }), detail.complete);
+                return true;
+                break;
```

Comment:
Same comment about return + break. It'd be nicer in an `if , else if ` set of blocks.

Reply:
I removed the breaks but kept the case statement in line with my other comment.